osu.Game/Database/RealmAccess.cs

```diff
@@ -986,6 +991,64 @@ private void applyMigrationsForVersion(Migration migration, ulong targetVersion)
 
                     break;
                 }
+
+                case 32:
+                {
+                    foreach (var score in migration.NewRealm.All<ScoreInfo>())
+                    {
+                        if (!score.IsLegacyScore || !score.Ruleset.IsLegacyRuleset())
+                            continue;
+
+                        string? replayFilename = score.Files.FirstOrDefault(f => f.Filename.EndsWith(@".osr", StringComparison.InvariantCultureIgnoreCase))?.File.GetStoragePath();
+                        if (replayFilename == null)
+                            continue;
+
+                        try
+                        {
+                            using (var stream = files.Store.GetStream(replayFilename))
+                            {
+                                if (stream == null)
+                                    continue;
+
+                                // Trimmed down logic from LegacyScoreDecoder to extract the mods bitmask.
+                                using (SerializationReader sr = new SerializationReader(stream))
+                                {
+                                    sr.ReadByte(); // Ruleset.
+                                    sr.ReadInt32(); // Version.
+                                    sr.ReadString(); // Beatmap hash.
+                                    sr.ReadString(); // Username.
+                                    sr.ReadString(); // MD5Hash.
+                                    sr.ReadUInt16(); // Count300.
+                                    sr.ReadUInt16(); // Count100.
+                                    sr.ReadUInt16(); // Count50.
+                                    sr.ReadUInt16(); // CountGeki.
+                                    sr.ReadUInt16(); // CountKatu.
+                                    sr.ReadUInt16(); // CountMiss.
+
+                                    // we should have this in LegacyTotalScore already, but if we're reading through this anyways...
+                                    int totalScore = sr.ReadInt32();
+
+                                    sr.ReadUInt16(); // Max combo.
+                                    sr.ReadBoolean(); // Perfect.
+
+                                    var legacyMods = (LegacyMods)sr.ReadInt32();
+
+                                    if (!legacyMods.HasFlagFast(LegacyMods.ScoreV2) || score.APIMods.Any(mod => mod.Acronym == @"SV2"))
+                                        continue;
+
+                                    score.APIMods = score.APIMods.Append(new APIMod(new ModScoreV2())).ToArray();
```

Comment:
`APIMods` is used here for speed. `Mods` is [orders of magnitude slower](https://gist.github.com/bdach/29f4e84525a7b40274540b2eff80e29c) due to relying on reflection.